### DISCUSSION TITLE
Validator and NamingStrategy inconsistent

### DISCUSSION THREAD

**Discussion by ad-on-is** [adonisjs/core#2734] [Help]
I'm using the `CamelCaseNamingStrategy` for serialization only. So, my tableNames and columNames are all in **snake_case** (default), while my model properties and JSON-responses are all in **camelCase**. However, if I send a POST/PUT/PATCH request with camelCased data and want to validate it, I receive an error.

User.ts

```js
class User extends BaseModel {
   column()
   public firstName: string
   // ...
}
```

migration.ts

```js
this.schema.createTable('users', (table) => {
	table.increments('id')
	table.timestamps(true)
	table.string('first_name')
})
```

## Getting the user

Serialized response data GET /users/:id

```
{"firstName": "John Foo"}
```
## Updating a user

Request data PUT /users/:id (does not work)
```
{"firstName": "Johnny Bar"}
```

UsersController.ts update() method

```js
 // ...
  const userSchema = schema.create({
    firstName: schema.string,
  })

  const validated= await request.validate({ schema: userSchema, existsStrict: true })
  await User.query().where({ userId: auth.user?.id }).update(validated)
  return response.ok()
  
// ....
```

This throws an error `update "users" set "firstName" = $1 where "user_id" = $2 - column "firstName" of relation "users" does not exist`.

Is there a way to have consistency in this case? I mean seanding camelCased data, which automatically gets snake_cased when performing DB operations?

But, if I send the data like this, it works.

Request data PUT /users/:id (works)
```
{"first_name": "Johnny Bar"}
```

**Comment by arthurcvm** [comment]
Hi! In error message your field is "firstNAme" and you send "firstName", please verify your migration to fix it.

**Reply by ad-on-is** [reply]
No no... it was a typo on my end when writing this code as a minimal example.

**Reply by arthurcvm** [reply]
Ok, please send your migration code.

**Reply by ad-on-is** [reply]
just updated it in my question.

**Reply by Vishwas-verma** [reply]
Hey, in your controller code where you are validating firstName it should be like **firstName: schema.string()**.
You have forgotten to place brackets in line **firstName:schema.string**

Hope it works.

**Comment by thetutlage** [comment]
Can you please share the code that runs the update query?

**Reply by ad-on-is** [reply]
I just appended it to my question.

Also, sending the following request data via PUT /users/:id works.
```
{"first_name": "Johnny Bar"}
```

**Reply by thetutlage** [reply]
You can sending the value in the `snake_case` format but your validator is looking for the field name in `camelCase`. So you should be getting a validation error at first place. 

Also, the naming strategy is for the models and not for the validator. So do not expect automatic key conversion with validator

**Reply by ad-on-is** [reply]
Aah, I see... I thought there was a setting for the naming strategy to do automatic key conversions, which then would apply to the validator. Thx for clarification.

**Reply by thetutlage** [reply]
Yup it doesn't. However I have plans to introduce it in validator too. But it needs some planning, so will take some time
